### DISCUSSION TITLE
Getting a reference to HTML attributes inside NVDA

### DISCUSSION THREAD

**Discussion by ehteshamulhaque71** [nvaccess/nvda#14716] [Q&A]
Is it possible to get a reference to HTML attributes (default and custom) inside NVDA? Let's say I have a tag `<textarea id="myID" rows="5" data-current-line="2"></textarea>`. Is it possible to get a list of these attributes and their values inside NVDA for this object using an add-on so that I can perform some action using the attributes?

I found that `api.getFocusObject().devInfo` gives `IAccessible2 attributes`, which lists `id` and `class`, but I could not find the remaining attributes like `rows` and the custom attribute `data-current-line`.

**Comment by seanbudd** [comment]
NVDA only has access to the accessibility tree via IAccessible2 attributes, full HTML markup is not passed through to the accessibility APIs consumed by screen readers.

https://developer.mozilla.org/en-US/docs/Glossary/Accessibility_tree
